### PR TITLE
BIG-3938 `qless-py-worker` gentle roll-outs

### PR DESCRIPTION
**Pull request by lindseyreno**
Worked with @dlecocq on this.  We were unable to write a test that mimics the behavior we saw but we did test this solution on a testing crawler.

@neilmb @b4hand

### CODE REVIEW

File: qless/workers/forking.py [hunk 1]
```diff
@@ -96,8 +98,12 @@ def run(self):
                 else:  # pragma: no cover
                     with Worker.sandbox(sandbox):
                         os.chdir(sandbox)
-                        self.spawn(sandbox=sandbox).run()
-                        exit(0)
+                        try:
+                           self.spawn(sandbox=sandbox).run()
+                        except Exception as exc:
```
Comment (dlecocq) [inline qless/workers/forking.py]: I think we don't need this `except` block

**Comment by dlecocq** [comment]
Small nit, but otherwise LGTM

File: test/test_worker.py [hunk 2]
```diff
@@ -20,9 +20,12 @@ def setUp(self):
 
     def test_proctitle(self):
         '''Make sure we can get / set the process title'''
-        before = Worker.title()
-        Worker.title('Foo')
-        self.assertNotEqual(before, Worker.title())
+        try:
```
Comment (dlecocq) [inline test/test_worker.py]:
```python
try:
    import setproctitle
    before = Worker.title()
    ...
except ImportError:
    pass
```

That way we only run this test if the environment actually has `setproctitle` available to it. Then we can re-enable the `except ImportError` block in `__init__`

Comment (lindseyreno) [inline test/test_worker.py]: You were too quick!

**Comment by lindseyreno** [comment]
Ok, now you can take a look.  Everything is passing now.

**Comment by dlecocq** [comment]
LGTM

File: test/test_worker.py [hunk 3]
```diff
@@ -20,9 +20,13 @@ def setUp(self):
...
+            import setproctitle
+            before = Worker.title()
+            Worker.title('Foo')
+            self.assertNotEqual(before, Worker.title())
+        except ImportError:
+            pass
```
Comment (neilmb) [inline test/test_worker.py]: Or if you want to be explicit about it, you could skip the test from the except block.
```python
except ImportError:
    self.skipTest('setproctitle not available')
```

**Comment by neilmb** [comment]
LGTM with one piffle.

File: test/test_worker.py (hunk 3 above)
Comment (dlecocq) [inline test/test_worker.py]: Good idea!

Comment (lindseyreno) [inline test/test_worker.py]: I like that too! Adding that.

**Pull request merged**